aws/resource_aws_kinesis_firehose_delivery_stream.go

```diff
@@ -1348,6 +1447,78 @@ func resourceAwsKinesisFirehoseDeliveryStream() *schema.Resource {
 				},
 			},
 
+			"http_endpoint_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"url": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.All(
+								validation.StringLenBetween(1, 1000),
+								validation.StringMatch(regexp.MustCompile(`^https://.*$`), ""),
+							),
+						},
+
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ValidateFunc: validation.All(
+								validation.StringLenBetween(1, 256),
+							),
+						},
+
+						"access_key": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringLenBetween(0, 4096),
+						},
+
+						"role_arn": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validateArn,
+						},
+
+						"s3_backup_mode": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      firehose.HttpEndpointS3BackupModeFailedDataOnly,
+							ValidateFunc: validation.StringInSlice(firehose.HttpEndpointS3BackupMode_Values(), false),
+						},
+
+						"retry_duration": {
+							Type:         schema.TypeInt,
+							Optional:     true,
+							Default:      3600,
+							ValidateFunc: validation.IntBetween(0, 7200),
+						},
```

Comment:
This attribute does not appear to be handled in the expand functions or documented. When running the current acceptance test with fixing the name length, the API seems to be setting this to 300 by default with it unconfigured:

```
=== CONT  TestAccAWSKinesisFirehoseDeliveryStream_HTTPEndpointConfigUpdates
    resource_aws_kinesis_firehose_delivery_stream_test.go:1174: Step 1/3 error: After applying this test step, the plan was not empty.
        stdout:


        An execution plan has been generated and is shown below.
        Resource actions are indicated with the following symbols:
          ~ update in-place

        Terraform will perform the following actions:

          # aws_kinesis_firehose_delivery_stream.test will be updated in-place
          ~ resource "aws_kinesis_firehose_delivery_stream" "test" {
                arn            = "arn:aws:firehose:us-west-2:--OMITTED--:deliverystream/terraform-kinesis-firehose-httpendpoint-1205375700746293683"
                destination    = "http_endpoint"
                destination_id = "destinationId-000000000001"
                id             = "arn:aws:firehose:us-west-2:--OMITTED--:deliverystream/terraform-kinesis-firehose-httpendpoint-1205375700746293683"
                name           = "terraform-kinesis-firehose-httpendpoint-1205375700746293683"
                version_id     = "1"

              ~ http_endpoint_configuration {
                    buffering_interval = 300
                    buffering_size     = 5
                    name               = "HTTP_test"
                  ~ retry_duration     = 300 -> 3600
                    role_arn           = "arn:aws:iam::--OMITTED--:role/tf_acctest_firehose_delivery_role_1205375700746293683"
                    s3_backup_mode     = "FailedDataOnly"
                    url                = "https://input-test.com:443"

                    cloudwatch_logging_options {
                        enabled = false
                    }

                    processing_configuration {
                        enabled = false
                    }

                    request_configuration {
                        content_encoding = "NONE"
                    }
                }

                s3_configuration {
                    bucket_arn         = "arn:aws:s3:::tf-test-bucket-1205375700746293683"
                    buffer_interval    = 300
                    buffer_size        = 5
                    compression_format = "UNCOMPRESSED"
                    role_arn           = "arn:aws:iam::--OMITTED--:role/tf_acctest_firehose_delivery_role_1205375700746293683"

                    cloudwatch_logging_options {
                        enabled = false
                    }
                }

                server_side_encryption {
                    enabled  = false
                    key_type = "AWS_OWNED_CMK"
                }
            }

        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccAWSKinesisFirehoseDeliveryStream_HTTPEndpointConfigUpdates (94.61s)
```

```suggestion
						"retry_duration": {
							Type:         schema.TypeInt,
							Optional:     true,
							Default:      300,
							ValidateFunc: validation.IntBetween(0, 7200),
						},
```

Reply:
This attribute is still not handled in the expand functions or documented. Will handle on merge to save the back and forth.